plugins/inputs/reader/reader.go

```diff
@@ -0,0 +1,95 @@
+package reader
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/globpath"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+)
+
+type Reader struct {
+	Filepaths     []string `toml:"files"`
+	FromBeginning bool
+	parser        parsers.Parser
+
+	Filenames []string
```

Comment:
Make this a unexported field with a lowercase letter, otherwise you can set this from the config file which is undesired.